tests/project.rs

```diff
@@ -2731,3 +2731,55 @@ contract ContractTest {
     compiled.assert_success();
     assert!(compiled.find_first("Contract").is_some());
 }
+
+// This is a repro and a regression test for https://github.com/foundry-rs/compilers/pull/45
+#[test]
+fn dirty_files_discovery() {
```

Comment:
Wonder if we can have another test or two, that explicitly check for proper recompilation of only dirty files? And maybe also a test like this, but it's all libraries but the top contract.